Wikipedia/Code/SectionEditorViewController.swift

```diff
@@ -145,10 +146,40 @@ class SectionEditorViewController: UIViewController {
         }
         let message = WMFLocalizedString("wikitext-downloading", value: "Loading content...", comment: "Alert text shown when obtaining latest revision of the section being edited")
         WMFAlertManager.sharedInstance.showAlert(message, sticky: true, dismissPreviousAlerts: true)
-        let manager = QueuesSingleton.sharedInstance()?.sectionWikiTextDownloadManager
-        QueuesSingleton.sharedInstance()?.sectionWikiTextDownloadManager.wmf_cancelAllTasks {
-            let _ = WikiTextSectionFetcher(andFetchWikiTextFor: section, with: manager, thenNotify: self)
+        sectionFetcher.fetch(section) { (result, error) in
+            if let error = error {
+                WMFAlertManager.sharedInstance.showErrorAlert(error as NSError, sticky: true, dismissPreviousAlerts: true)
```

Comment:
should this and the other `showAlert` be dispatched to the main thread?

Reply:
yup, fixed